app/src/main/java/org/oppia/app/options/OptionsFragment.kt

```diff
@@ -0,0 +1,261 @@
+package org.oppia.app.options
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.Transformations
+import androidx.preference.Preference
+import androidx.preference.PreferenceFragmentCompat
+import androidx.preference.PreferenceManager
+import org.oppia.app.R
+import org.oppia.app.drawer.KEY_NAVIGATION_PROFILE_ID
+import org.oppia.app.model.AppLanguage
+import org.oppia.app.model.AudioLanguage
+import org.oppia.app.model.Profile
+import org.oppia.app.model.ProfileId
+import org.oppia.app.model.StoryTextSize
+import org.oppia.domain.profile.ProfileManagementController
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.logging.Logger
+import javax.inject.Inject
+
+const val KEY_MESSAGE_STORY_TEXT_SIZE = "Text Size"
+const val KEY_MESSAGE_APP_LANGUAGE = "App Language"
+const val KEY_MESSAGE_AUDIO_LANGUAGE = "Audio Language"
+
+class OptionsFragment @Inject constructor(
+  private val activity: AppCompatActivity,
+  private val profileManagementController: ProfileManagementController,
+  private val logger: Logger
+) : PreferenceFragmentCompat() {
+  private var internalProfileId: Int = -1
+  private lateinit var profileId: ProfileId
+  private var storyTextSize = StoryTextSize.SMALL_TEXT_SIZE
+  private var appLanguage = AppLanguage.ENGLISH_APP_LANGUAGE
+  private var audioLanguage = AudioLanguage.NO_AUDIO
+
+  override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+    setPreferencesFromResource(R.xml.basic_preference, rootKey)
+    internalProfileId = activity.intent.getIntExtra(KEY_NAVIGATION_PROFILE_ID, -1)
```

Comment:
if the id is `-1` how will the application respond ?

Reply:
It is getting id from Profilechooser. It cannot have -1.

Reply:
ok.